src/System.Private.ServiceModel/tools/PRService/pr.ashx

```diff
@@ -60,26 +62,35 @@ public class PullRequestHandler : IHttpHandler
             context.Response.Write(HttpUtility.HtmlEncode(result));
             return;
         }
-        else
+
+        success = false;
+
+        // Check inputs to the script. PR has precedence
+        uint pr;
+        if (!string.IsNullOrEmpty(prString) && uint.TryParse(prString, out pr))
         {
-            // Check inputs to the script
-            uint pr;
-            if (!string.IsNullOrEmpty(prString) && uint.TryParse(prString, out pr))
-            {
-                success = SyncToPr(pr, result);
-            }
-            else
+            success = SyncToPr(pr, result);
+        }
+        else if (!string.IsNullOrEmpty(branchString))
```

Comment:
For a request like `pr.ashx?pr=123&branch=1.0.0-rc2`, we will actually check out pr=123 from master branch anyway and ignore branch=1.0.0-rc2. This does not feel right. I see what you really mean is that `pr` and `branch` should not be specified at the same time.

My recommendation: just like how we implement any other APIs, move all the parameter validation logic to the beginning of this function including null/empty check, tryparse, no-coexistence, master branch only and so on. This avoids calling `CheckPrerequisites` and `CleanupMergedBranches` to just find out the input parameters are invalid. This will also make your code here much nicer just like any other blocks you had.
```
success = pr>0 ? SyncToPr(pr, result) : SyncToBranch(branchString, result);
if (!success) {
    //... ...
}
```